File: solutions_5636311922769920_1/Python/afoo/q4.py

def get_ceil(K,C):
	if K%C==0:
		return K/C;
	return K/C+1;
T = input();

for Ti in range(T):
	line = raw_input();
	#print(line); 
	[K,C,S] = map(int, line.split());
	num = get_ceil(K,C);
	if S<num:
		ans = "IMPOSSIBLE";
		print("Case #{0}: {1}".format(Ti+1,ans));
		continue;
		
	print("Case #{0}: ".format(Ti+1)),
	for j in range(num):
		ans = 0;
		for k in range(C):
			if j*C+k>=K:
				break;
			ans *= K;
			ans += j*C+k;
		
		print(" {0}".format(ans+1)),
	print("");

File: solutions_5636311922769920_0/C++/afoo/q4.cxx
#include <stdio.h>

    int get_ceil(int a, int b) {
        if (a%b==0) 
            return a/b;
        return a/b+1;
    }
int main()
{
    int T;
    int K,C,S;
    scanf("%d", &T);
    for (int i=1;i<=T;i++) {
        scanf("%d %d %d", &K,&C,&S);
        printf("Case #%d: ", i);
        int num = get_ceil(K,C);
        if (S<num) {
            printf("IMPOSSIBLE\n");
            continue;
        }
//        printf("IMPOSSIBLE\n");
/*
        for (int j=0;j<num;j++) {
            int ans =0;
            for (int k=0;k<C && j*C+k<K ;k++) {
                ans *= K;
                ans += j*C+k;
            }
            printf("%d ", ans+1);
        }
*/
        for (int j=1;j<=S;j++) {
            printf("%d ", j);
        }
        printf("\n");
        

    }
    return 0;
}


